x-pack/plugins/reporting/export_types/csv_from_savedobject/types.d.ts

```diff
@@ -76,12 +76,18 @@ export interface SavedObject {
   references: SavedObjectReference[];
 }
 
+/* This object is passed to different helpers in different parts of the code
+   - packages/kbn-es-query/src/es_query/build_es_query
+   - x-pack/plugins/reporting/export_types/csv/server/lib/field_format_map
+   The structure has redundant parts and json-parsed / json-unparsed versions of the same data
+ */
 export interface IndexPatternSavedObject {
   title: string;
   timeFieldName: string;
   fields: any[];
-  fieldFormatMap: {
-    [key: string]: { id: string; params: { pattern: string } };
+  attributes: {
+    fieldFormatMap: string;
+    fields: string;
```

Comment:
There was some structure tweaks to this type made, after realizing that `buildEsQuery` and `getFieldFormatMap` both require an index pattern object, but expect different structure :\